providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env python3
+"""prime offload test module."""
+
+import sys
+import threading
+import subprocess
+import time
+import re
+import json
+import argparse
+import logging
+from enum import IntEnum
+
+
+class PrimeOffloaderError(IntEnum):
+    """
+    A class used to define PrimeOffloader Error code
+
+    Attributes
```

Comment:
I like to use the Sphinx way of writing docstrings, such as the one used in checkbox internals, for instance:

https://github.com/canonical/checkbox/blob/main/checkbox-ng/plainbox/impl/session/state.py#L229

You can find more info here: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html